### PR TITLE
feat(pkg/scale): `VaryingDataTypeValue` String method

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

As proposed by Eclesio in https://github.com/ChainSafe/gossamer/pull/2965#discussion_r1028345455

1. ~`VaryingDataTypeValue` interface is now composed of `fmt.Stringer`~
2. Add missing `String() string` methods to types needing to implement `VaryingDataTypeValue`
3. Add `String() string` method on `VaryingDataType` struct
4. Add `String() string` method on `VaryingDataTypeSlice` struct

## Issues

A few more steps towards #1945 

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM! Happy with log improvements as well as relying on type systems instead of runtime type checks!! 🚀

File: pkg/scale/varying_data_type.go [hunk 1]
```diff
@@ -5,11 +5,13 @@ package scale
 
 import (
 	"fmt"
+	"strings"
 )
 
 // VaryingDataTypeValue is used to represent scale encodable types of an associated VaryingDataType
 type VaryingDataTypeValue interface {
 	Index() uint
+	String() string
```
Comment (timwu20) [inline pkg/scale/varying_data_type.go:14]: I don't think we need to enforce this for all `VaryingDataTypeValue` implementations.

**Review by timwu20 — CHANGES_REQUESTED** [review]

File: pkg/scale/varying_data_type.go (hunk 1 above)
Comment (jimjbrettj) [reply pkg/scale/varying_data_type.go:14]: I would say I agree with this

**Review by jimjbrettj — APPROVED** [review]
Aside from Tims comment this looks good to me

File: pkg/scale/varying_data_type.go (hunk 1 above)
Comment (qdm12) [reply pkg/scale/varying_data_type.go]: Good point, removed from interface and changed `VaryingDataType`'s String method to be:

```go
func (vdt VaryingDataType) String() string {
	if vdt.value == nil {
		return "VaryingDataType(nil)"
	}
	stringer, ok := vdt.value.(fmt.Stringer)
	if !ok {
		return fmt.Sprintf("VaryingDataType(%v)", vdt.value)
	}
	return stringer.String()
}
```

Does that work with you? 🤔

**Review by timwu20 — APPROVED** [review]

**Pull request merged**